src/NuGet.Core/NuGet.Packaging/Rules/MissingReadmeRule.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+
+namespace NuGet.Packaging.Rules
+{
+    public class MissingReadmeRule : IPackageRule
+    {
+        public string MessageFormat { get; }
+
+        public MissingReadmeRule(string messageFormat)
+        {
+            MessageFormat = messageFormat ?? throw new ArgumentNullException(nameof(messageFormat));
+        }
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var nuspecReader = builder?.NuspecReader;
+            var readmeMetadata = nuspecReader?.GetReadme();
+
+            if (readmeMetadata == null)
+            {
+                yield return PackagingLogMessage.CreateMessage(
+                    string.Format(CultureInfo.CurrentCulture, Environment.NewLine + MessageFormat),
+                    LogLevel.Minimal);
```

Comment:
This message doesn't have an NU code. I believe all other pack warnings have NU codes. I know you're putting a URL in the message itself. Was it intentional to be different than the other rules and not have an NU code?

I'm thinking, what happens when the customer opts into "generate package on build", so this warning comes up as part of build. Will it appear in Visual Studio's Error List window?  If so, an NU code is very useful, because it's clickable, taking the customer (hopefuly) directly to the docs page.

Reply:
Thanks! This is not a warning or an error, so there is no NU code.
It's a message similar with [Log_PackageCommandSuccess](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L202-L204)

Reply:
I disagree. The Log_PackageCommandSuccess reports the outcome of the action, whereas this PR's message is an analyzer message displayed when a condition is met. Just like all the other warnings and errors.

Let's say there's a customer who really doesn't want a readme in their package.  How can they suppress this message? If it had an NU code, they could put that NU code in a `NoWarn`.

It's out of scope for this PR, but since pack warnings are really a lot like Roslyn Analyzers, I wish we could have `.editorconfig` settings to allow customers to change [severity levels](https://learn.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2022#configure-severity-levels). From a customer experience point of view, it's really really good, whereas MSBuild NoWarn, WarningAsError, WarningNotAsError is much more limited, and the language (MSBuild) to configure it is much more clunky.

Reply:
As the description in https://github.com/NuGet/Home/issues/12070, `TreatWarningsAsError` isn't an uncommon thing, so warnings in these cases can be disruptive.
So if we add this info as warning, then yes, customers could suppress this message, but also, some customers might encounter build break for this best practice validation, if they use `TreatWarningsAsError`.

Reply:
Thinking about it again, I think the cost of implementing severity levels in pack is pretty low, we'd just need a forward looking design.